Prog/vacant-lots-montgomery.Rmd

```diff
@@ -0,0 +1,929 @@
+
+```{r file-check}
+
+if (!file.exists(paste0("L:/Libraries/RegHsg/Doc/zoning-codes/",
+                              filepath,
+                              "-zoning-code.csv"))) {
+  
+  stop("Move zoning file for jurisdiction to L:/Libraries/RegHsg/Doc/zoning-codes directory")
+}
+
+```
+
+```{r read-zoning-csv}
+z <- cols(
+  Jur = col_character(),
+  Code = col_character(),
+  `Zone` = col_character(),
+  `Type of Dwelling` = col_character(),
+  `Site Area` = col_number(),
+  `Site area per dwelling unit (sq. ft.) (minimum)` = col_number(),
+  `Lot area` = col_number(),
+  `Lot area per dwelling unit (sq. ft.) (minimum)` = col_number(),
+  `Lot Width (average minimum, ft)` = col_character(),
+  `Lot Width building line (average minimum, ft)` = col_character(),
+  `Height (ft)` = col_double(),
+  `Site coverage maximum` = col_number(),
+  `Lot coverage maximum` = col_number(),
+  `Floor area minimum` = col_double(),
+  `Maximum Density of Floor-Area-Ratio (FAR)` = col_number(),
+  `Setbacks (front)` = col_character(),
+  `Side (ft)` = col_character(),
+  `Rear (ft)` = col_character(),
+  `Density (units/acre)` = col_number(),
+  `Massing Units/Row` = col_number(),
+  `Open Space minimum percentage` = col_number(),
+  Comments = col_character()
+)
+
+zoningcode <- read_csv(paste0("L:/Libraries/RegHsg/Doc/zoning-codes/",
+                              filepath,
+                              "-zoning-code.csv"),
+                       col_types = z)
+
+names(zoningcode) <- 
+  c("juristdiction", "zoning_code", "zone", "dwelling_type",
+    "site_area_min", "sitearea_per_unit", "lot_area_min", "lotarea_per_unit",
+    "lot_width","lot_width_building", "height_ft",
+    "site_coverage_max", "lot_coverage_max", "floor_area_min", "far_max",
+    "setbacks_front", "setbacks_side",
+    "setbacks_rear", "density", "massing", "open_space", "comments")
+
+# fix variables read in as whole numbers (should be percent)
+zoningcode <- zoningcode %>% 
+  mutate_at(vars(lot_coverage_max, site_coverage_max, open_space),
+            ~ . / 100)
+rm(z)
+
+```
+
+### Creating dwelling_type hierarchy
+
+For the vacant lots analysis, we want to determine the `dwelling_type` that allows for the most dense construction for that zone. 
+
+First, we look at all the options, and filter out all the observations marked as "other", since we cannot be sure what property type that refers to. We will also filter out categories that are dependent on adjacent lot lines.
+
+```{r tab-dwellings}
+
+zoningcode %>% 
+  count(dwelling_type, juristdiction) %>% 
+  knitr::kable()
+```
+
+Count the number of unique zoning codes to make sure we won't get rid of any codes entirely.
+
+```{r count-zoning}
+length(unique(zoningcode$zoning_code))
+
+```
+
+## Create zoning for each jur so we can split the analysis
+
+```{r zoning by juristdiction}
+
+zoning_montgomery <- zoningcode %>% 
+  filter(juristdiction == "Montgomery")
+
+zoning_rockville <- zoningcode %>% 
```

Comment:
@maggiesu0725 added a csv write out if you want to use it - once I run it it will be rockville-zoning-code-clean.csv and gaithersburg-zoning-code-clean.csv in L:/Libraries/RegHsg/Doc/zoning-codes